zend_abstract_interface/functions/CMakeLists.txt

```diff
@@ -0,0 +1,28 @@
+add_library(zai_functions "${PHP_VERSION_DIRECTORY}/functions.c")
+
+target_include_directories(zai_functions PUBLIC
+                                         $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}>
+                                         $<INSTALL_INTERFACE:include>)
+
+target_compile_features(zai_functions PUBLIC c_std_99)
+
+target_link_libraries(zai_functions PUBLIC "${PHP_LIB}" Zai::Sandbox)
+
+set_target_properties(zai_functions PROPERTIES
+                                    EXPORT_NAME Functions
+                                    VERSION ${PROJECT_VERSION})
+
+add_library(Zai::Functions ALIAS zai_functions)
+
+if (${BUILD_ZAI_TESTING})
+  add_subdirectory(tests)
+endif()
+
+# This copies the include files when `install` is ran
+# TODO: How to make this zai/functions.h?
```

Comment:
To answer your question on making it `zai/functions.h`, you would set the `DESTINATION` to `"${CMAKE_INSTALL_INCLUDEDIR}/zai"` but to make that _also_ work prior to install (the common case for us) you need to adjust the directory structure so that it lives in `zai/functions.h`.

Reply:
Ah yes, I remember our conversation on this a while back. I believe we decided to not put all the headers in the same directory for some reason, but I can't remember the reason. I think we decided to keep it `thing/thing.h` until we could figure out a proper solution that we liked. Since this is also something that impacts `libddprof`, I would love to sync up with you on this again to iron out a direction we want to go.